tools/integration_tests/buffered_read/helpers_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buffered_read
+
+import (
+	"bytes"
+	"context"
+	"os"
+	"path"
+	"syscall"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/googlecloudplatform/gcsfuse/v3/tools/integration_tests/util/client"
+	"github.com/googlecloudplatform/gcsfuse/v3/tools/integration_tests/util/log_parser/json_parser/read_logs"
+	"github.com/googlecloudplatform/gcsfuse/v3/tools/integration_tests/util/operations"
+	"github.com/googlecloudplatform/gcsfuse/v3/tools/integration_tests/util/setup"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Expected is a helper struct that stores list of attributes to be validated from logs.
+type Expected struct {
+	StartTimeStampSeconds int64
+	EndTimeStampSeconds   int64
+	BucketName            string
+	ObjectName            string
+}
+
+func readFileAndValidate(ctx context.Context, storageClient *storage.Client, testDir, fileName string, readFullFile bool, offset int64, chunkSizeToRead int64, t *testing.T) *Expected {
+	expected := &Expected{
+		StartTimeStampSeconds: time.Now().Unix(),
+		BucketName:            setup.TestBucket(),
+		ObjectName:            path.Join(path.Base(testDir), fileName),
+	}
+	if setup.DynamicBucketMounted() != "" {
+		expected.BucketName = setup.DynamicBucketMounted()
+	}
+
+	var content []byte
+	var err error
+
+	if readFullFile {
```

Comment:
Can we utilize [this](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/tools/integration_tests/read_cache/helpers_test.go#L46) only by putting it in common client.go file?